dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/srt/SRTServletRequest.java

```diff
@@ -2873,17 +2845,16 @@ public void pushParameterStack()
         if (WCCustomProperties.CHECK_REQUEST_OBJECT_IN_USE){
             checkRequestObjectInUse();
         }
-        SRTServletRequestThreadData reqData = SRTServletRequestThreadData.getInstance();
-        if (reqData.getParameters() == null)
+        if (this._parameters == null)
         {
-            reqData.pushParameterStack(null);
+            _paramStack.push(null);
```

Comment:
reqData.popParameter moved the top-of-stack to _parameters
`    public void popParameterStack()
    {
        try
        {
            _parameters = (Hashtable) _paramStack.pop();`
from where it could then be retrieved by getParameters()

I'm not sure about pushing the null, but maybe that is so that after the include/forward completes, the following pop will return the null, rather than whatever was top-of-stack before the include/forward started?

Reply:
agreed, have changed it as per your suggestion